### PR TITLE
feat(remix-vercel): add support for `@vercel/node` v2

### PR DESCRIPTION
**Pull request by styfle**
Fixes a peerDependency issue with npm

```
> npx create-remix
? Where would you like to create your app? ./my-remix-app
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix if you're unsure; it's easy to change deployment targets. Vercel
? Do you want me to run `npm install`? Yes
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: undefined@undefined
npm ERR! Found: @vercel/node@2.0.1
npm ERR! node_modules/@vercel/node
npm ERR!   @vercel/node@"^2.0.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @vercel/node@"^1.8.3" from @remix-run/vercel@1.5.1
npm ERR! node_modules/@remix-run/vercel
npm ERR!   @remix-run/vercel@"^1.5.1" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

Command failed: npm install
```

### CODE REVIEW

File: packages/remix-vercel/package.json [hunk 1]
```diff
@@ -16,13 +16,10 @@
...
   "devDependencies": {
     "@types/supertest": "^2.0.10",
-    "@vercel/node": "^1.8.3",
+    "@vercel/node": "^2.0.1",
     "node-mocks-http": "^1.10.1",
     "supertest": "^6.1.5"
   },
-  "peerDependencies": {
-    "@vercel/node": "^1.8.3"
-  },
```
Comment (MichaelDeBoey) [inline packages/remix-vercel/package.json:25]: We should keep `@vercel/node` in `peerDependencies`.
We should however support having v2.

```suggestion
  "peerDependencies": {
    "@vercel/node": "^1.8.3 || ^2.0.0"
  },
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]
I should have thought about this when doing #3381 🙈

File: packages/remix-vercel/package.json (hunk 1 above)
Comment (styfle) [reply packages/remix-vercel/package.json:25]: Its only use to import a type. Why is it needed for `peerDependencies`?

Comment (MichaelDeBoey) [reply packages/remix-vercel/package.json:25]: @styfle The imported types are used in both `GetLoadContextFunction` & `RequestHandler`, which are both exported by the adapter, so it should at least be in `peerDependencies`

**Comment by kentcdodds** [comment]
@mjackson is responsible for the adapters, so I've assigned it to him :)

**Comment by MichaelDeBoey** [comment]
@styfle Since build & tests are failing on some internal `@vercel/node` things, you'll probably need to redo the `yarn.lock` changes.

**Comment by MichaelDeBoey** [comment]
@styfle Could you please also sign the CLA?
Without signing it, we can't merge this PR.

**Comment by styfle** [comment]
I'll have to check with my employer first

**Comment by MichaelDeBoey** [comment]
If it's not possible, we can re-open @mcansh's #3420 instead.

**Comment by styfle** [comment]
Thats fine

**Comment by MichaelDeBoey** [comment]
@styfle You mean it's OK to re-open & merge that one?
Or you want to wait for an answer of your employer first?

**Comment by styfle** [comment]
Its fine to re-open the other PR

**Comment by MichaelDeBoey** [comment]
Re-opened #3420 👍

**Pull request closed without merging**